hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -278,6 +278,210 @@ def test_failure_reason(reason):
     '''
     return reason
 
+def test_mount_attrs(mount_name,attribute):
+    '''
+    Ensure that a given directory is mounted with appropriate attributes
+    '''
+    output = _execute_shell_command('mount | grep ' + mount_name)
+    if output.strip() == '':
+        return 'No match'
+    elif attribute not in output:
+        return output
+    else:
+        return True
+
+def service_not_enabled(service_name):
+    '''
+    Ensure that the given service is not running on the system. Return False if the service is running.
+    Return True otherwise
+    '''
+    output = _execute_shell_command('systemctl is-enabled ' + service_name + ' 2>/dev/null')
+    print output
+    if output.strip() == 'enabled':
+        return output
+    else:
+        return True
+
+def ensure_ssh_disabled_within_containers():
+    '''
+    Ensure that ssh deamon is not running within any container(s). 
+    '''
+    instances_raw = _execute_shell_command("docker ps --quiet")
+    instances_array = instances_raw.split("\n")
+    return_list = []
+    return_value = True
+    for instance in instances_array:
+        command = 'docker exec ' + instance + ' ps -el | grep "sshd$"'
+        output = _execute_shell_command(command)
+        if output.strip() != '':
+            return_list.append(output)
+            return_value = False
+    if return_value == False:
+        return str(return_list)
+    else:
+        return True
+
+def privileged_ports_not_mapped():
+    '''
+    Ensure that any privileged port ( below 1024 ) is not mapped within a docker container
+    '''
+    command = "docker ps --quiet | xargs docker inspect --format '{{ .Id }}: Ports={{ .NetworkSettings.Ports }}'"
+    output = _execute_shell_command(command)
+    ports = re.findall("HostPort:(\d+)]",output)
+    return_list = []
+    return_value = True
+    for port in ports:
+        if int(port) < 1024:
+            return_value  = False
+            return_list.append(port)
+
+    if return_value == False:
+        return return_list
+    else:
+        return True
+
+def ensure_sticky_bit_on_writable_directories():
```

Comment:
There is already a function as sticky_bit_on_world_writable_dirs. Please check that.